From d81ece07875ec7b2ff24e8d3e0c69d7206b09c2c Mon Sep 17 00:00:00 2001
From: Eliad Peller <eliad@wizery.com>
Date: Tue, 15 Mar 2011 13:06:12 +0200
Subject: [PATCH 18/75] wl12xx: avoid redundant join/unjoin on interface reconfiguration

ieee80211_reconfig() sets most of the "changed" flags regardless
of the actual change (e.g. BSS_CHANGED_ASSOC will be set even if
the interface is still not associated). in this case the driver
will issue some unneeded commands.

Avoid some of these redundant commands by considering the actual
state in addition to the "changed" flag.

(there seem to be more redundant configuration commands, but
they shouldn't harm. however, redundant join/unjoin have some
potential side effects)
---
 drivers/net/wireless/wl12xx/main.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/drivers/net/wireless/wl12xx/main.c b/drivers/net/wireless/wl12xx/main.c
index 5bfe998..b10a588 100644
--- a/drivers/net/wireless/wl12xx/main.c
+++ b/drivers/net/wireless/wl12xx/main.c
@@ -3020,7 +3020,8 @@ static void wl1271_bss_info_changed_sta(struct wl1271 *wl,
 			wl1271_configure_filters(wl, 0);
 
 			/* Need to update the BSSID (for filtering etc) */
-			do_join = true;
+			if (bss_conf->assoc)
+				do_join = true;
 		}
 	}
 
-- 
1.7.0.4

